Guides/Best Practices and FAQ.md

```diff
@@ -0,0 +1,57 @@
+# Best Practices and FAQs
+
+This guide provides notes and details on best practices in using `IGListKit`, general tips, and answers to FAQs.
+
+## Best Practices
+
+- We recommend adding an assert to check [`-isKindOfClass:`](https://developer.apple.com/reference/objectivec/1418956-nsobject/1418511-iskindofclass) on the object you receive in [`-didUpdateToObject:`](https://instagram.github.io/IGListKit/Protocols/IGListSectionType.html#/c:objc(pl)IGListSectionType(im)didUpdateToObject:) in your section controllers. 
+This makes it easy to track down easily-overlooked mistakes in your [`IGListAdapaterDataSource`](https://instagram.github.io/IGListKit/Protocols/IGListAdapterDataSource.html#/c:objc(pl)IGListAdapterDataSource(im)listAdapter:sectionControllerForObject:) implementation. 
+If this assert is ever hit, that means `IGListKit` has sent your section controller the incorrect type of object. 
+This would only happen if your objects provide *non-unique* diff identifiers. 
+
+```objective-c
+- (void)didUpdateToObject:(id)object {
+    NSParameterAssert([object isKindOfClass:[MyModelClass class]]);
+    _myModel = object;
+}
+```
+
+- Make sure your [`-diffIdentifier`](https://instagram.github.io/IGListKit/Protocols/IGListDiffable.html#/c:objc(pl)IGListDiffable(im)diffIdentifier) implementation returns a **unique identifier** for each object.
+
+- We highly recommend using single-item sections when possible. That is, each section only has 1 cell. Each section controller manages a single model and a single cell. This gives you the greatest amount of flexibility, modularity, and re-use for your components.
```

Comment:
Do we want to recommend this? I think your point is not to use a single section controller w/ tons of variable items, right? We probably don't want to discourage multi-cell section controllers in general.